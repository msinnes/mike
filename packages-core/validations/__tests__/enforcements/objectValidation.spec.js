const objectValidationEnforcement = require('../../enforcements/objectValidation');
const Validation = require('../../classes/Validation');
const ArrayValidation = require('../../classes/ArrayValidation');
const ObjectValidation = require('../../classes/ObjectValidation');
const MapValidation = require('../../classes/MapValidation');

const testValidation = new Validation(() => {});
const testArrayValidation = new ArrayValidation(testValidation);
const testObjectValidation = new ObjectValidation({
  fieldOne: testValidation,
});
const testMapValidation = new MapValidation({
  values: testValidation,
});

describe('objectValidationEnforcement', () => {
  it('should be a function', () => {
    expect(objectValidationEnforcement).toBeDefined();
    expect(objectValidationEnforcement).toBeInstanceOf(Function);
  });

  it('should not throw an error', () => {
    expect(() => {
      objectValidationEnforcement({});
    }).not.toThrow();
    expect(() => {
      objectValidationEnforcement({
        field1: testValidation,
      });
    }).not.toThrow();
    expect(() => {
      objectValidationEnforcement({
        field1: testValidation,
        field2: testArrayValidation,
      });
    }).not.toThrow();
    expect(() => {
      objectValidationEnforcement({
        field1: testMapValidation,
        field2: testObjectValidation,
      });
    }).not.toThrow();
  });

  it('should throw an error', () => {
    expect(() => {
      objectValidationEnforcement();
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      objectValidationEnforcement([]);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      objectValidationEnforcement('string');
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      objectValidationEnforcement(() => {});
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      objectValidationEnforcement(() => {
        objectValidationEnforcement({
          field1: () => {},
        });
      });
    }).toThrowErrorMatchingSnapshot();
  });
});
