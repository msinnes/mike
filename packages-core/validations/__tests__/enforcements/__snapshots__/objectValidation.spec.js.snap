// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectValidationEnforcement should throw an error 1`] = `"InternalError: An Object Validation should be an object and all fields should be validations"`;

exports[`objectValidationEnforcement should throw an error 2`] = `"InternalError: An Object Validation should be an object and all fields should be validations"`;

exports[`objectValidationEnforcement should throw an error 3`] = `"InternalError: An Object Validation should be an object and all fields should be validations"`;

exports[`objectValidationEnforcement should throw an error 4`] = `"InternalError: An Object Validation should be an object and all fields should be validations"`;

exports[`objectValidationEnforcement should throw an error 5`] = `"InternalError: An Object Validation should be an object and all fields should be validations"`;
